app/models/grda_warehouse/ch_enrollment.rb

```diff
@@ -179,10 +179,14 @@ def self.terminating?(sym)
 
     # Line 1 (3.08)
     def self.disabling_condition(enrollment)
-      result = if is_no?(enrollment.DisablingCondition)
-        :no
-      elsif dk_or_r_or_missing(enrollment.DisablingCondition)
-        dk_or_r_or_missing(enrollment.DisablingCondition)
+      # if we don't have a yes from DisablingCondition, look to the disability entry records
+      # if none of the disabilities are indefinite and impairing, then return the DisablingCondition
+      result = if (is_no?(enrollment.DisablingCondition) || dk_or_r_or_missing(enrollment.DisablingCondition)) && enrollment.disabilities_at_entry&.map(&:indefinite_and_impairs?)&.none?(true)
+        if is_no?(enrollment.DisablingCondition)
+          :no
+        elsif dk_or_r_or_missing(enrollment.DisablingCondition)
+          dk_or_r_or_missing(enrollment.DisablingCondition)
```

Comment:
Because of the guard above, I think that if `is_no?` returns false, this will always be true... 
```suggestion
        else
          dk_or_r_or_missing(enrollment.DisablingCondition)
```
If you prefer the belt and suspenders, I'd like an explicit `else` here